tracee-ebpf/tracee/tracee.go

```diff
@@ -1085,9 +1122,16 @@ func (t *Tracee) processEvent(ctx *context, args map[argTag]interface{}) error {
 					//TODO: remove dead pid from cache
 					continue
 				}
-				//don't capture same file twice unless it was modified
+
 				sourceFileCtime := sourceFileStat.Sys().(*syscall.Stat_t).Ctim.Nano()
 				capturedFileID := fmt.Sprintf("%d:%s", ctx.MntID, sourceFilePath)
+
+				// create an in-memory profile
+				if t.config.Capture.Profile {
```

Comment:
Why not to put this after the check of already captured file below?
This will prevent unnecessary processing for duplications

Reply:
You're right. I've updated this.

Reply:
Actually we need this for a different reason. If we were only profiling the captured files once your suggestion will be valid. But in our case we need to also include the number of times a file was executed for which we cannot skip over duplicates.

Reply:
Well, when I wrote this comment, "Times" was not part of the profiling...